### PR TITLE
Beginner Friendly Example in README

### PR DESCRIPTION
**Pull request by mjohns39**
Added a more beginner friendly example and explanation to the README to help them bridge the gap from vanilla logging to log4cats

### CODE REVIEW

**Comment by mjohns39** [comment]
Opening this following a discussion in TypeLevel Discord server:  https://discord.com/channels/632277896739946517/632278585700384799/1024710118023114824

File: README.md [hunk 1]
```diff
@@ -18,6 +18,24 @@ libraryDependencies ++= Seq(
 )
 ```
 
+## Why log4cats?
+
+Short answer is you don't really need log4cats if you already have logging setup in your app.
```
Comment (danicheg) [inline README.md:23]: > – Why log4cats?
> – Short answer is you don't really need log4cats...

I'm sorry, but that totally creased me 😆 
Can we change this to something less disappointing for maintainers? 🙇🏻‍♂️

Comment (mjohns39) [reply README.md]: Done.  Let me know what you think of the new version :)

Comment (mjohns39) [reply README.md]: P.S.  I squashed the commits

File: README.md [hunk 2]
```diff
@@ -18,6 +18,23 @@ libraryDependencies ++= Seq(
 )
 ```
 
+## Why log4cats?
+
```
Comment (lorandszakacs) [inline README.md:22]: I'd also include another sentence of the sort:
```
Logging is oftimes an overlooked side-effect, which under the hood can range from writing to a mutable queue, writing to disk, outputing to the console, or sometimes even doing network I/O! To correctly deal with these kinds of side-effects we have to ensure they are properly wrapped in `IO`, see the [cats-effect docs](https://typelevel.org/cats-effect/docs/concepts#side-effects) for more details.
```

And then tie it in with the sentence you wrote.

Now I realize this essentially doubles the amounts of words :sweat_smile: it definitely requires _some_ editing :D, and I'm not clinging to any particular form or formulation. Just want to point readers to the fact that there is documentation around dealing with side-effects.

Comment (mjohns39) [reply README.md:22]: I see your point.  The purpose of this example was targeted towards new learners and people possibly in a hybrid state that want to be using cats-effect with their existing app.  The goal was to give something easy to understand, but also valid with the cats-effect ethos.  So I don't think I should include it in the "Why log4cats?" section...but I will add it further down

Comment (mjohns39) [reply README.md:22]: Pushed change.  Let me know what you think?

Comment (lorandszakacs) [reply README.md:22]: Nice!

**Review by lorandszakacs — APPROVED** [review]
Double checked for typos.

Really like the `## Wait...why log4cats again?` section :seedling:

**Pull request merged**